include/deal.II/dofs/dof_handler.h

```diff
@@ -418,19 +438,21 @@ class DoFHandler : public Subscriptor
    * finite element indices may be used on different cells, and the default
    * index there corresponds to an invalid value.
    */
-  static const unsigned int default_fe_index = 0;
+  static const unsigned int default_fe_index =
+    numbers::invalid_unsigned_int; // TODO
 
   /**
    * Standard constructor, not initializing any data. After constructing an
    * object with this constructor, use initialize() to make a valid
    * DoFHandler.
    */
-  DoFHandler();
+  DoFHandler(const bool is_hp_dof_handler = false);
```

Comment:
I must admit that I do not like the name `is_hp_dof_handler` because it reflects more the evolution of this flag rather than the functionality (keeping the topic aside whether we might or might not keep it like this in the long run). I prefer the name `hp_capability_enabled` or something like that.
```suggestion
  DoFHandler(const bool hp_capability_enabled = false);
```